src/index.ts

```diff
@@ -120,10 +124,21 @@ export class Client {
             let theError = error;
             await console.error(theError);
             // @ts-ignore
-            // console.error(theError.type);
-            // @ts-ignore
             if (theError.type == "invalid-json"){
                 await this.delay(1000);
+
```

Comment:
suggestion:

this should be then extracted (as a method) , which then internally tries this in a loop

you could use either `for` if you have a max amount of tries like 10 times
or you could use `while (true) { including the delays and stuff`  until you find something and then just use return